src/wolframite/base/convert.clj

```diff
@@ -94,12 +118,18 @@
             (throw (Exception. (str "Symbols passed to Mathematica must be alphanumeric (apart from forward slashes and dollar signs). Passed: " s)))
             (proto/expr (jlink-instance/get) :Expr/SYMBOL s)))))))
 
+(defn- convert-non-simple-list [elms opts]
+  (let [converted-parts (map #(cond-> % (dispatch %) (convert opts)) elms)]
+    (if (every? (partial proto/expr? (jlink-instance/get)) converted-parts)
+      (proto/expr (jlink-instance/get)
+                  (cons (convert 'List opts)
+                        converted-parts))
+      (convert (to-array converted-parts) opts))))

```

Comment:
this is what we used to do before in such cases, and will in turn invoke the :null convert